### PR TITLE
Relocatable Compilation support for Method Handles in Eclipse OpenJ9

### PR DESCRIPTION
**Pull request by dsouzai**
This PR contains changes necessary to support the Method Handle feature in Eclipse OpenJ9 (https://github.com/eclipse/openj9/issues/4850), namely:

- Adding a new relo type, which is used to relocate special methods known to the VM.
- Ensuring consistency between the isInterpreted flag between the resolved method and resolved method symbol.

### CODE REVIEW

File: compiler/il/OMRResolvedMethodSymbol.cpp [hunk 1]
```diff
@@ -157,15 +157,17 @@ OMR::ResolvedMethodSymbol::ResolvedMethodSymbol(TR_ResolvedMethod * method, TR::
    //
    if ((_methodIndex > JITTED_METHOD_INDEX && !_resolvedMethod->isSameMethod(comp->getJittedMethodSymbol()->getResolvedMethod()))
        || comp->isDLT()
-       || (comp->getOption(TR_UseSymbolValidationManager) && comp->compileRelocatableCode()))
+       || comp->compileRelocatableCode())
```
Comment (0xdaryl) [inline compiler/il/OMRResolvedMethodSymbol.cpp:160]: Please fix copyright header.

Comment (dsouzai) [reply compiler/il/OMRResolvedMethodSymbol.cpp:160]: Didn't OMR have at one point a copyright header checker? Do you know why it was removed?

**Comment by dsouzai** [comment]
Copyright updated.

**Comment by 0xdaryl** [comment]
Jenkins build all

File: compiler/il/OMRResolvedMethodSymbol.cpp (hunk 1 above)
Comment (0xdaryl) [reply compiler/il/OMRResolvedMethodSymbol.cpp:160]: Never implemented: #2254

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**